Bottom: c88ede17b15d9b5b62b15019b223036cd3bb5c5b
Top:    44169cbc47b9a6f74da012e3d6b4d8f99e53e3d2
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2008-12-30 10:40:03 -0500

Refresh of jds-audio

---

diff --git a/sound/soc/codecs/tas5504.c b/sound/soc/codecs/tas5504.c
index 72d7c12..929baf8 100644
--- a/sound/soc/codecs/tas5504.c
+++ b/sound/soc/codecs/tas5504.c
@@ -291,7 +291,8 @@ static int tas5504_reg_write(struct snd_soc_codec *codec, unsigned int reg,
  * Digital Audio Interface Operations
  */
 static int tas5504_hw_params(struct snd_pcm_substream *substream,
-			   struct snd_pcm_hw_params *params)
+			   struct snd_pcm_hw_params *params,
+			   struct snd_soc_dai *dai)
 {
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
 	struct snd_soc_device *socdev = rtd->socdev;
@@ -334,8 +335,6 @@ struct snd_soc_dai tas5504_dai = {
 	},
 	.ops = {
 		.hw_params = tas5504_hw_params,
-	},
-	.dai_ops = {
 		.digital_mute = tas5504_mute,
 	},
 };
@@ -726,7 +725,7 @@ static int tas5504_probe(struct platform_device *pdev)
 
 	/* CODEC is setup, we can register the card now */
 	dev_dbg(&pdev->dev, "Registering card\n");
-	ret = snd_soc_register_card(socdev);
+	ret = snd_soc_init_card(socdev);
 	if (ret < 0) {
 		dev_err(&pdev->dev, "tas5504: failed to register card\n");
 		goto card_err;
